pkg/yang/identity_test.go

```diff
@@ -0,0 +1,415 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yang
+
+import "testing"
+
+// inputModule is a mock input YANG module.
+type inputModule struct {
+	name    string // The filename of the YANG module.
+	content string // The contents of the YANG module.
+}
+
+type idrefOut struct {
+	module string   // The module that the identityref is within.
+	name   string   // The name of the identityref.
+	values []string // Names of the identities that the identityref relates to.
+}
+
+// identityOut is the output for a particular identity within the test case.
+type identityOut struct {
+	module   string   // The module that the identity is within.
+	name     string   // The name of the identity.
+	baseName string   // The base of the identity as a string.
+	values   []string // The string names of derived identities.
+}
+
+// identityTestCase is a test case for a module which contains identities.
+type identityTestCase struct {
+	name       string
+	in         []inputModule // The set of input modules for the test
+	identities []identityOut // Slice of the identity values expected
+	idrefs     []idrefOut    // Slice of identityref results expected
+	err        string        // Test case error string
+}
+
+// Test cases for basic identity extraction.
+var basicTestCases = []identityTestCase{
+	identityTestCase{
+		name: "basic-test-case-1: Check identity is found in module.",
+		in: []inputModule{
+			inputModule{
+				name: "idtest-one",
+				content: `
+module idtest-one {
+  namespace "urn:idone";
+  prefix "idone";
+
+  identity TEST_ID;
+}
```

Comment:
It is up to you, but this might be more readable if  you indent the text to be one tab in from content:

```
                        content: `
                                module idtest-one {
                                        namespace "urn:idone";
                                ...
                        `,
                },

```